src/Container.php

```diff
@@ -64,13 +72,14 @@ public function add($alias, $concrete = null, $singleton = false)
             return null;
         }
 
-        // get a definition of the item
-        $this->items[$alias]['singleton'] = (boolean) $singleton;
-
+        // we need to build a definition
         $factory    = $this->getDefinitionFactory();
         $definition = $factory($alias, $concrete, $this);
 
-        $this->items[$alias]['definition'] = $definition;
+        $this->items[$alias] = [
+            'definition' => $definition,
+            'singleton'  => (boolean) $singleton
```

Comment:
This is actually preference, `bool` and `boolean` on casting work in exactly the same way and I don't see any definition in the spec to say one shouldn't be used, if we were talking about a type check then fine, but it's casting so makes no difference